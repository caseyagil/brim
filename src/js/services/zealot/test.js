/* @flow */
import zealot from "./"

describe("search api", () => {
  let client = zealot.client("localhost:9867")
  test("the search params", () => {
    expect(client.inspect().search("count()")).toEqual({
      method: "POST",
      path: "/search?format=zjson",
      body: expect.any(String)
    })
  })

  test("search defaults", () => {
    client.searchDefaults({
      space: "corelight",
      from: "Jan 1, 2019",
      to: "Jan 10, 2019"
    })
  })
})

describe("spaces api", () => {
  let client = zealot.client("localhost:9867")

  test("the spaces list api", () => {
    expect(client.inspect().spaces.list()).toEqual({
      method: "GET",
      path: "/space"
    })
  })

  test("the spaces get api", () => {
    expect(client.inspect().spaces.get("awesome")).toEqual({
      method: "GET",
      path: "/space/awesome"
    })
  })

  test("create a space without a name", () => {
    expect(client.inspect().spaces.create()).toEqual({
      method: "POST",
      path: "/space"
    })
  })

  test("delete a space with a name", () => {
    expect(client.inspect().spaces.delete("my-space")).toEqual({
      method: "DELETE",
      path: "/space/my-space"
    })
  })
})

describe("ingest api", () => {
  let client = zealot.client("localhost:9867")

  test("get pcaps", () => {
    expect(client.inspect().pcaps.get({space: "hi"})).toEqual({
      method: "GET",
      path: "/space/hi/pcap"
    })
  })

  test("post pcaps", () => {
    expect(
      client.inspect().pcaps.post({space: "default", file: "~/my/capture.pcap"})
    ).toEqual({
      method: "POST",
      path: "/space/default/pcap",
      body: expect.any(String)
    })
  })

  test("post logs with no types", () => {
    let paths = ["~/zeek/1.log", "~/zeek/2.log"]
    expect(client.inspect().logs.post({space: "default", paths})).toEqual({
      method: "POST",
      path: "/space/default/log",
      body: JSON.stringify({paths})
    })
  })

  test("post logs json with default types", () => {
    let paths = ["~/zeek/1.log", "~/zeek/2.log"]
    let data = client
      .inspect()
      .logs.post({space: "default", paths, types: "default"})

    expect(JSON.parse(data.body).json_type_config).toEqual(expect.any(Object))
  })

  test("post logs with types as a string ", () => {
    let paths = ["~/zeek/1.log", "~/zeek/2.log"]
    let types = '{"descriptors": [], "rules": []}'
    let data = client.inspect().logs.post({space: "default", paths, types})

    expect(JSON.parse(data.body).json_type_config).toEqual({
      descriptors: [],
      rules: []
    })
  })

  test("post logs with types as an object", () => {
    let paths = ["~/zeek/1.log", "~/zeek/2.log"]
    let types = {descriptors: [], rules: []}
    let data = client.inspect().logs.post({space: "default", paths, types})

    expect(JSON.parse(data.body).json_type_config).toEqual({
      descriptors: [],
      rules: []
    })
  })
})
