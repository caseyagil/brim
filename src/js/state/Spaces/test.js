/* @flow */
import type {SpaceDetailPayload} from "../../services/zealot/types"
import Spaces from "./"
import initTestStore from "../../test/initTestStore"

let store
beforeEach(() => {
  store = initTestStore()
})

let detail: SpaceDetailPayload = {
  name: "default",
  span: {
    ts: {sec: 1425564900, ns: 0},
    dur: {sec: 3352893, ns: 750000000}
  },
  pcap_support: true
}

test("setting the names", () => {
  let state = store.dispatchAll([
    Spaces.setNames("cluster1", ["default", "hq_integration"])
  ])

  expect(Spaces.names("cluster1")(state)).toEqual(["default", "hq_integration"])
})

test("space names removing", () => {
  let selector = Spaces.names("cluster1")
  let state = store.dispatchAll([
    Spaces.setNames("cluster1", ["default", "hq_integration"]),
    Spaces.setNames("cluster1", ["default"])
  ])

  expect(selector(state)).toEqual(["default"])
})

test("setting the space detail adds the defaults", () => {
  let state = store.dispatchAll([Spaces.setDetail("cluster1", detail)])

  expect(Spaces.get("cluster1", "default")(state)).toEqual({
    name: "default",
    pcap_support: true,
    min_time: {sec: 1425564900, ns: 0},
    max_time: {sec: 1428917793, ns: 750000000},
    ingest: {
      progress: null,
      warnings: [],
      snapshot: null
    }
  })
})

test("setting the ingest progress throws error if no space yet", () => {
  expect(() => {
    store.dispatchAll([Spaces.setIngestProgress("cluster1", detail.name, 0.5)])
  }).toThrow("No space exists with name: default")
})

test("setting the ingest progress", () => {
  let actions = Spaces.actionsFor("cluster1", detail.name)
  store.dispatchAll([actions.create(), actions.setIngestProgress(0.5)])

  let value = Spaces.getIngestProgress(
    "cluster1",
    detail.name
  )(store.getState())

  expect(value).toEqual(0.5)
})

test("getting the spaces with details, others not", () => {
  let state = store.dispatchAll([
    Spaces.setNames("cluster1", ["space-a", "space-b"]),
    Spaces.setDetail("cluster1", {...detail, name: "space-a"})
  ])
  let spaces = Spaces.getSpaces("cluster1")(state)

  expect(spaces).toEqual([
    {
      name: "space-a",
      max_time: {ns: 750000000, sec: 1428917793},
      min_time: {ns: 0, sec: 1425564900},
      pcap_support: true,
      ingest: {warnings: [], progress: null, snapshot: null}
    },
    {
      name: "space-b",
      max_time: {ns: 0, sec: 0},
      min_time: {ns: 0, sec: 0},
      pcap_support: false,
      ingest: {warnings: [], progress: null, snapshot: null}
    }
  ])
})

test("only cares about spaces actions", () => {
  store.dispatch({type: "NON_SPACE"})
  expect(Spaces.raw(store.getState())).toEqual({})
})

test("ingest warnings", () => {
  let actions = Spaces.actionsFor("cluster1", detail.name)
  let state = store.dispatchAll([
    actions.create(),
    actions.appendIngestWarning("Problem 1"),
    actions.appendIngestWarning("Problem 2")
  ])

  expect(Spaces.getIngestWarnings("cluster1", detail.name)(state)).toEqual([
    "Problem 1",
    "Problem 2"
  ])
})

test("clear warnings", () => {
  let actions = Spaces.actionsFor("cluster1", detail.name)
  let state = store.dispatchAll([
    actions.create(),
    actions.appendIngestWarning("Problem 1"),
    actions.appendIngestWarning("Problem 2"),
    actions.clearIngestWarnings()
  ])

  expect(Spaces.getIngestWarnings("cluster1", detail.name)(state)).toEqual([])
})

test("remove space", () => {
  let actions = Spaces.actionsFor("cluster1", detail.name)

  let state = store.dispatchAll([actions.create(), actions.remove()])

  expect(Spaces.getSpaces("cluster1")(state)).toEqual([])
})

test("setting the spanshot counter", () => {
  let actions = Spaces.actionsFor("cluster1", detail.name)

  let state = store.dispatchAll([
    actions.create(),
    actions.setIngestSnapshot(1)
  ])

  expect(Spaces.getIngestSnapshot("cluster1", detail.name)(state)).toBe(1)
})
