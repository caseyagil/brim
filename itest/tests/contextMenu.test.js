/* @flow */

import {basename} from "path"

import {
  click,
  newAppInstance,
  pcapIngestSample,
  rightClick,
  searchDisplay,
  setSpan,
  startApp,
  startSearch,
  waitForResults,
  writeSearch
} from "../lib/app.js"
import {handleError, stdTest} from "../lib/jest.js"
import {selectors} from "../../src/js/test/integration"

describe("Test search mods via right-clicks", () => {
  let app
  let testIdx = 0

  beforeAll(async () => {
    app = newAppInstance(basename(__filename), ++testIdx)
    await startApp(app)
    await pcapIngestSample(app)
    await setSpan(app, "Whole Space")
  })

  afterAll(async () => {
    if (app && app.isRunning()) {
      await app.stop()
    }
  })

  beforeEach(async () => {
    await writeSearch(app, "")
    await startSearch(app)
    await waitForResults(app)
  })

  stdTest("count by _path; sort asc", (done) => {
    rightClick(app, selectors.viewer.resultCellContaining("capture_loss"))
      .then(async () => {
        await click(app, selectors.viewer.contextMenuItem("Count by field"))
        await rightClick(app, selectors.viewer.resultCellContaining("dns"))
        await click(app, selectors.viewer.contextMenuItem("Sort A...Z"))
        let results = await searchDisplay(app)
        expect(results).toMatchSnapshot()
        done()
      })
      .catch((err) => {
        handleError(app, err, done)
      })
  })
})
