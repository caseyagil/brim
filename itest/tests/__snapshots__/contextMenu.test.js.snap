// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test search mods via right-clicks count by _path; sort asc 1`] = `
Array [
  "_path",
  "count",
  "capture_loss",
  "1",
  "conn",
  "19",
  "dns",
  "2",
  "files",
  "4",
  "ssl",
  "1",
  "stats",
  "1",
  "weird",
  "1",
  "x509",
  "1",
]
`;
